Vienna/Sources/Main window/FoldersTree.m

```diff
@@ -1601,4 +1290,311 @@ -(void)dealloc
+                NSInteger folderId = [dbManager addRSSFolder:feedTitle underParent:parentId afterChild:predecessorId subscriptionURL:feedURL];
+                if (feedDescription != nil) {
+                    [dbManager setDescription:feedDescription forFolder:folderId];
+                }
+                if (feedHomePage != nil) {
+                    [dbManager setHomePage:feedHomePage forFolder:folderId];
+                }
+                if (folderId > 0) {
+                    folderToSelect = folderId;
+                }
+                ++childIndex;
+            }
+        }
+
+        // If parent was a group, expand it now
+        if (parentId != VNAFolderTypeRoot)
+            [self.outlineView expandItem:[self.rootNode nodeFromID:parentId]];
+        
+        // Select a new folder
+        if (folderToSelect > 0)
+            [self selectFolder:folderToSelect];
+        
+        return YES;
+    }
+    if ([type isEqualToString:@"WebURLsWithTitlesPboardType"])
+    {
+        Database * dbManager = [Database sharedManager];
+        NSArray * webURLsWithTitles = [pb propertyListForType:type];
+        NSArray * arrayOfURLs = webURLsWithTitles[0];
+        NSArray * arrayOfTitles = webURLsWithTitles[1];
+        NSInteger count = arrayOfURLs.count;
+        NSInteger index;
+        
+        __block NSInteger folderToSelect = -1;
+        for (index = 0; index < count; ++index)
+        {
+            NSString * feedTitle = arrayOfTitles[index];
+            NSString * feedURL = arrayOfURLs[index];
+            NSURL * draggedURL = [NSURL URLWithString:feedURL];
+            if ((draggedURL.scheme != nil) && [draggedURL.scheme isEqualToString:@"feed"])
+                feedURL = [NSString stringWithFormat:@"http:%@", draggedURL.resourceSpecifier];
+            
+            if ([dbManager folderFromFeedURL:feedURL] == nil)
+            {
+                NSInteger predecessorId = (childIndex > 0) ? [node childByIndex:(childIndex - 1)].nodeId : 0;
+                NSInteger newFolderId = [dbManager addRSSFolder:feedTitle underParent:parentId afterChild:predecessorId subscriptionURL:feedURL];
+                if (newFolderId > 0) {
+                    folderToSelect = newFolderId;
+                }
+                ++childIndex;
+            }
+        }
+        
+        // If parent was a group, expand it now
+        if (parentId != VNAFolderTypeRoot)
+            [self.outlineView expandItem:[self.rootNode nodeFromID:parentId]];
+        
+        // Select a new folder
+        if (folderToSelect > 0)
+            [self selectFolder:folderToSelect];
+        
+        return YES;
+    }
+    return NO;
+}
+
+-(NSView *)outlineView:(NSOutlineView *)outlineView viewForTableColumn:(NSTableColumn *)tableColumn item:(id)item {
+    FolderTreeCellView *folderTreeCellView = nil;
+    if ([tableColumn.identifier isEqualToString:@"folderColumns"])
```

Comment:
This is the only case, so to avoid indentation negate it and make it an early exit